pkg/api/middlewares.go

```diff
@@ -23,70 +23,48 @@ import (
 	"runtime/debug"
 	"time"
 
-	"github.com/megaease/easegress/pkg/common"
+	"github.com/go-chi/chi/v5/middleware"
 	"github.com/megaease/easegress/pkg/logger"
-
-	"github.com/kataras/iris/context"
 )
 
-func newAPILogger() func(context.Context) {
-	return func(ctx context.Context) {
-		var (
-			method            string
-			remoteAddr        string
-			path              string
-			code              int
-			bodyBytesReceived int64
-			bodyBytesSent     int64
-			startTime         time.Time
-			processTime       time.Duration
-		)
-
-		startTime = common.Now()
-		ctx.Next()
-		processTime = common.Now().Sub(startTime)
-
-		method = ctx.Method()
-		remoteAddr = ctx.RemoteAddr()
-		path = ctx.Path()
-		code = ctx.GetStatusCode()
-		bodyBytesReceived = ctx.GetContentLength()
-		bodyBytesSent = int64(ctx.ResponseWriter().Written())
+func (s *Server) newAPILogger(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ww := middleware.NewWrapResponseWriter(w, r.ProtoMajor)
 
-		logger.APIAccess(method, remoteAddr, path, code,
-			bodyBytesReceived, bodyBytesSent,
-			startTime, processTime)
-	}
+		t1 := time.Now()
+		defer func() {
+			logger.APIAccess(r.Method, r.RemoteAddr, r.URL.Path, ww.Status(),
+				r.Response.ContentLength, int64(ww.BytesWritten()),
```

Comment:
Directly using `r.Response` will cause panic in runtime. Here is the stack
```bash
 ^[[31mERROR^[[0m        api/middlewares.go:48   recover from /apis/v1/status/members, err: runtim
e error: invalid memory address or nil pointer dereference, stack trace:
goroutine 2603 [running]:
runtime/debug.Stack(0xc00300b650, 0x2035320, 0x2c9c6f0)
        /usr/local/go/src/runtime/debug/stack.go:24 +0x9f
github.com/megaease/easegress/pkg/api.(*Server).newRecoverer.func1.1(0xc000221300, 0x24266a0, 0xc00055a9a0)
        /Users/benja/programmer/github/tmp/easegress/pkg/api/middlewares.go:49 +0xba
panic(0x2035320, 0x2c9c6f0)
        /usr/local/go/src/runtime/panic.go:965 +0x1b9
github.com/megaease/easegress/pkg/api.(*Server).newAPILogger.func1.1(0xc000221300, 0x24389b8, 0xc00020b9c0, 0xc02a7defc8b09818, 0
x317fa75a8a, 0x2cbf420)
        /Users/benja/programmer/github/tmp/easegress/pkg/api/middlewares.go:37 +0xc4
github.com/megaease/easegress/pkg/api.(*Server).newAPILogger.func1(0x24266a0, 0xc00055a9a0, 0xc000221300)
        /Users/benja/programmer/github/tmp/easegress/pkg/api/middlewares.go:41 +0x17b
net/http.HandlerFunc.ServeHTTP(0xc0005c0600, 0x24266a0, 0xc00055a9a0, 0xc000221300)
        /usr/local/go/src/net/http/server.go:2069 +0x44
github.com/megaease/easegress/pkg/api.(*Server).newRecoverer.func1(0x24266a0, 0xc00055a9a0, 0xc000221300)
        /Users/benja/programmer/github/tmp/easegress/pkg/api/middlewares.go:58 +0x83
net/http.HandlerFunc.ServeHTTP(0xc0005c0618, 0x24266a0, 0xc00055a9a0, 0xc000221300)
        /usr/local/go/src/net/http/server.go:2069 +0x44
github.com/megaease/easegress/pkg/api.(*Server).newConfigVersionAttacher.func1(0x24266a0, 0xc00055a9a0, 0xc000221300)
        /Users/benja/programmer/github/tmp/easegress/pkg/api/middlewares.go:68 +0x1f2
net/http.HandlerFunc.ServeHTTP(0xc00063c520, 0x24266a0, 0xc00055a9a0, 0xc000221300)
        /usr/local/go/src/net/http/server.go:2069 +0x44
github.com/go-chi/chi/v5.(*Mux).ServeHTTP(0xc0003dcb40, 0x24266a0, 0xc00055a9a0, 0xc000221200)
        /Users/benja/programmer/go/pkg/mod/github.com/go-chi/chi/v5@v5.0.3/mux.go:88 +0x310
net/http.serverHandler.ServeHTTP(0xc0007c81c0, 0x24266a0, 0xc00055a9a0, 0xc000221200)
        /usr/local/go/src/net/http/server.go:2887 +0xa3
net/http.(*conn).serve(0xc00079ca00, 0x242bb38, 0xc0027830c0)
        /usr/local/go/src/net/http/server.go:1952 +0x8cd
created by net/http.(*Server).Serve
        /usr/local/go/src/net/http/server.go:3013 +0x39b
```
Cause `r.Response` can only be used in the `HTTP client redirect`[1] scenario. 
Plz form a cluster by using the scripts in the `easegress/example` directory and check if the new feature is runnable locally.

# Reference
[1] https://pkg.go.dev/net/http#Request.Response